src/libraries/System.Private.Uri/src/System/Uri.cs

```diff
@@ -4003,7 +3992,7 @@ private static unsafe ParsingError CheckSchemeSyntax(ReadOnlySpan<char> span, re
             ushort start = idx;
             newHost = null;
             bool justNormalized = false;
-            bool iriParsing = (IriParsing && IriParsingStatic(syntax)); // perf
+            bool iriParsing = IriParsingStatic(syntax); // perf
```

Comment:
What do these `// perf` comments mean?

Reply:
I have no idea. I am guessing they are referring to "caching" the IriParsingStatic result to a local

Reply:
Let's remove 'em.